libcaf_core/src/uri.cpp

```diff
@@ -101,6 +102,27 @@ std::string to_string(const uri& x) {
   return result;
 }
 
+std::string to_string(const uri::authority_type& x) {
+  std::string str;
+  if (!x.userinfo.empty()) {
+    detail::append_uri(str, x.userinfo);
+    str += '@';
+  }
+  auto addr = get_if<ip_address>(&x.host);
+  if (addr == nullptr) {
+    detail::append_uri(str, get<std::string>(x.host));
```

Comment:
It's unlikely that the `host` variant receives further types, but I'd still prefer either using `caf::visit` for the dispatching instead to get a compiler error if the definition of `host` ever changes or at least put a `CAF_ASSERT(holds_alternative<std::string>(x.host))` here.